apps/challenges/views.py

```diff
@@ -340,3 +358,221 @@ def challenge_phase_split_list(request, challenge_pk):
+                    if test_annotation_file:
+                        test_annotation_file_path = os.path.join("{}/{}/{}{}".format(BASE_LOCATION,
+                                                                                     unique_folder_name,
+                                                                                     extracted_folder_name,
+                                                                                     test_annotation_file))
+                    if os.path.isfile(test_annotation_file_path):
+                        challenge_test_annotation = open(test_annotation_file_path, 'rb')
+                        challenge_test_annotation_file = File(challenge_test_annotation)
+                        challenge_phase = ChallengePhase.objects.get(pk=challenge_phase.pk)
+                        challenge_phase.test_annotation.save(test_annotation_file,
+                                                             challenge_test_annotation_file,
+                                                             save=True)
+
+            # Create Dataset Splits
+            yaml_file_data_of_dataset_split = yaml_file_data['dataset_splits']
+            dataset_split_ids = []
+            for data in yaml_file_data_of_dataset_split:
+                serializer = DatasetSplitSerializer(data=data)
+                if serializer.is_valid():
+                    dataset_split = serializer.save()
+                    dataset_split_ids.append(dataset_split.pk)
+
+            # Create Challenge Phase Splits
+            yaml_file_data_of_challenge_phase_splits = yaml_file_data['challenge_phase_splits']
+            for data in yaml_file_data_of_challenge_phase_splits:
+                challenge_phase = challenge_phase_ids[data['challenge_phase_id']-1]
+                leaderboard = leaderboard_ids[data['leaderboard_id']-1]
+                dataset_split = dataset_split_ids[data['dataset_split_id']-1]
+                visibility = data['visibility']
+
+                data = {
+                    'challenge_phase': challenge_phase,
+                    'leaderboard': leaderboard,
+                    'dataset_split': dataset_split,
+                    'visibility': visibility
+                }
+
+                serializer = ZipChallengePhaseSplitSerializer(data=data)
+                if serializer.is_valid():
+                    serializer.save()
+        zip_config = ChallengeConfiguration.objects.get(pk=uploaded_zip_file.pk)
+
+        if zip_config:
+            zip_config.challenge = challenge
+            zip_config.save()
+            response_data = {'success': 'The Challenge is successfully created'}
+            return Response(response_data, status=status.HTTP_201_CREATED)
+    except:
+        response_data = {'error': 'Error in challenge creation. Please try again'}
+        return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    try:
```

Comment:
Use `shutil.rmtree` for deleting folder. Also logger are for your help. We can make them more meaningful by making them more specific like adding ids or other data. So here in case folder is successfully deleted, no need of logging the message but please log proper specific message in case it fails to remove.